src/S3Bucket/Components/RestoreButton.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <VTooltip location="start" :text="restoreError">
+    <template #activator="{ props: activatorProps }">
+      <VIcon
+        v-if="restoreError"
+        v-bind="activatorProps"
+        color="error"
+        size="x-small"
+        icon="mdi-alert-circle"
+        class="mt-1"
+      />
+    </template>
+  </VTooltip>
+  <VBtn icon="mdi-file-undo" title="Restore File" @click="restoreItem" />
+</template>
+
+<script setup>
+import { computed, inject, ref } from 'vue';
+import { convertObjectToFormData } from '../../helpers/axiosHelper';
+import { useBuckets } from '../../helpers/useBuckets';
+
+/**
+ * @typedef {Object} Props
+ * @property {Object} Props.item - The S3 Bucket file item
+ */
+/** @type {Props} */
+
+const props = defineProps({
+  item: {
+    type: Object,
+    default: () => {}
+  }
```

Comment:
It's generally better to provide a bit more structure for props than a single flexible input object like this. It look like you actually expect props like `key`, `path`, and `version_id`. Doing so will give you better deugging, documentation, and prop tracking.

[You can still bind objects to them simply too](https://vuejs.org/guide/components/props.html#prop-passing-details), just instead of doing this:
```
<RestoreButton :item="item.raw" />
```

you'd unpack the object into props like this:
```
<RestoreButton v-bind="item.raw" />
```

Reply:
so this won't work for restore button in particular because one of the props would have to be "key" and vue gets mad about that, if I try and call it specifically in defineProps
![Screenshot 2023-08-30 at 4 19 55 PM](https://github.com/CloudBoltSoftware/applet-s3-bucket-browser/assets/105660893/98176949-dd70-499e-bb2d-55ad4da197fb)

Also one of them would be `version_id` which isn't camel case and makes it mad.

I did split them out to make things clearer though